### PR TITLE
Add layer information (IfcPresentationLayerAssignment) to IfcConvert'…

### LINKED ISSUE #141: Add layer information to IfcConvert's XML output?
I've been noticing that the XML file has gained various new tidbits recently. Would the layer information also a good applicant to be included in the file?

### PR DESCRIPTION
**Pull request by Stinkfist0**
…s XML output. Closes #141

### CODE REVIEW

**Comment by Stinkfist0** [comment]
Oh, noticed an inconsistency: I'm prefixing the layer ID's with #, but this is not done for types' or properties' IDs. Maybe we should simply remove the `id` which is duplicate to `Name` in any case. I.e.
` <IfcPresentationLayerAssignment xlink:href="#A-WALL-MBNI"/>` would stay as is but the ` <IfcPresentationLayerAssignment id="#A-WALL-MBNI" Name="A-WALL-MBNI"/>` would become ` <IfcPresentationLayerAssignment Name="A-WALL-MBNI"/>`

File: src/ifcconvert/XmlSerializer.cpp [hunk 1]
```diff
@@ -328,10 +368,20 @@ void XmlSerializer::finalize() {
 		}
 	}
 
+    // Layer assignments
+    IfcPresentationLayerAssignment::list::ptr layer_assignments = file->entitiesByType<IfcPresentationLayerAssignment>();
+    for (IfcPresentationLayerAssignment::list::it it = layer_assignments->begin(); it != layer_assignments->end(); ++it) {
+        // IfcPresentationLayerAssignments don't have GUIDs (only optional Identifier) so use name as the ID
+        ptree node;
+        node.put("<xmlattr>.id", "#" + (*it)->Name());
```
Comment (aothms) [inline src/ifcconvert/XmlSerializer.cpp]: The `"#"` is not part of the `id` (so should be removed here), but as part of the `href` it signifies that the link points to a node in the same document. The xlink w3c spec is a bit too verbose to me now, but the example here [1] illustrates this.

[1] https://en.wikipedia.org/wiki/XLink#Simple_links

**Comment by aothms** [comment]
What about duplicate `IfcPresentationLayerAssignment.Name`s? There is nothing in the IFC spec that would argue they are unique. But XML id attributes better be unique. That's why I in the #141 suggested a set of string with layer names.

It might be a bit of a stretch, but I can imagine there might be files in which the layer assignment name is not unique. For example when multiple files get merged into one, or perhaps there are simply some exporters that emit the multiple assignments to the same string label.

**Comment by Stinkfist0** [comment]
Aye, will fix this shorty.

**Comment by Stinkfist0** [comment]
^

**Comment by aothms** [comment]
I don't think postfixing is necessary. The layer assignment name I think is what matters, not really the assignment instance.

**Comment by Stinkfist0** [comment]
OK, should be fixed now.

**Comment by Stinkfist0** [comment]
Hi. Does this look OK now?

**Comment by aothms** [comment]
Suppose so :) sorry for the delay and many thanks as always!

**Comment by Stinkfist0** [comment]
FYI, I just encountered a file that used IfcPresentationLayerWithStyle and this seemed to handle also them just fine:
```xml
<IfcPresentationLayerWithStyle id="something" Name="something" LayerOn="true" LayerFrozen="false" LayerBlocked="false"/>
...
<IfcPresentationLayerWithStyle xlink:href="#something"/>
```

**Pull request merged**